crates/re_arrow_store/src/store.rs

```diff
@@ -23,8 +23,13 @@ pub struct DataStoreConfig {
     /// to a specific timeline _and_ a specific entity.
     ///
     /// This effectively puts an upper bound on the number of rows that need to be sorted when an
-    /// indexed bucket gets out of order.
+    /// indexed bucket gets out of order (e.g. because of new insertions or a GC pass).
     /// This is a tradeoff: less rows means faster sorts at the cost of more metadata overhead.
+    /// In particular:
+    /// - Query performance scales inversely logarithmically to this number (i.e. it gets better
+    ///   the higher this number gets).
```

Comment:
How so? We do a BTree-serarch to find the correct bucket, and then a binary search within the bucket, so it is logarithmic either way, no?

Reply:
The number of rows within a bucket is generally bounded (ignoring extremely pathological cases), while the number of buckets is can grow indefinitely

Reply:
But the query time of a N buckets of size 1 is log(N), and the query time of 1 bucket of size N is also log(N), no?

So the query time should be independent of the the size of the buckets afaict (at least within a constant factor)